shard/shard.go

```diff
@@ -166,12 +117,28 @@ func (s *shard) CutoffNanos() int64 {
 
 func (s *shard) SetCutoffNanos(value int64) Shard {
 	if value == DefaultShardCutoffNanos {
```

Comment:
Add comments explaining the rationale behind this?

Reply:
Sure